### ISSUE TITLE
For the cmdlet Set-MgUserManagerByRef, the first parameter variant does not work as defined.

### ISSUE THREAD

**Issue by msewaweru** [microsoftgraph/msgraph-sdk-powershell#705]
The parameter variant indicates that only the UserId is a mandatory parameter. 
```powershell
Set-MgUserManagerByRef
   -UserId <String>
   [-AdditionalProperties <Hashtable>]
   [-PassThru]
   [-WhatIf]
   [-Confirm]
   [<CommonParameters>] 
```
Running this as is returns the error below.

```
Set-MgUserManagerByRef : An unexpected 'EndOfInput' node was found when reading from the JSON reader. A 'StartObject' node was expected.
At line:1 char:1
+ Set-MgUserManagerByRef -Userid 8a7c50d3-fcbd-4727-a889-8ab232dfea01
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: ({ UserId = 8a7c...ionJsonSchema }:<>f__AnonymousType47`2) [Set-MgUserManagerByRef_SetExpanded], RestException`1
    + FullyQualifiedErrorId : Request_BadRequest,Microsoft.Graph.PowerShell.Cmdlets.SetMgUserManagerByRef_SetExpanded
``` 
[AB#9951](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9951)

**Comment by peombwa** [comment]
Thanks for surfacing this. This is due to us leaking OData constructs to PowerShell users. We need to provide a better way of representing [/$ref](http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_EntityIdsandEntityReferences) request bodies. In this case, the example lacks the necessary request body required to make a request. `-AdditionalProperties` should always be optional.

**Comment by peombwa** [comment]
New-MgDirectoryRoleMemberByRef, New-MgGroupMemberByRef, New-MgGroupMemberOfByRef, and Set-MgUserLicense are also affected by this.

**Comment by davehope** [comment]
Just discovered this prior to reporting a new bug (Get-MgTeamGroupByRef).

Get-MgTeamGroupByRef includes the literal value $ref" in its GET request.

```
GET https://graph.microsoft.com/v1.0/teams/xxx/group/$ref HTTP/1.1
SdkVersion: Graph-powershell-Microsoft.Graph.Authentication-1.6.0, Graph-dotnet-1.25.1
FeatureFlag: 00000047
Cache-Control: no-store, no-cache
Accept-Encoding: gzip
Host: graph.microsoft.com
```

Is there any update on when this may be reviewed? - Seems to occur with 1.6.1 and 1.7.0

**Comment by msewaweru** [comment]
New-MgEntitlementManagementAccessPackage,  New-MgEntitlementManagementAccessPackageResourceRequest also affected.

**Comment by peombwa** [comment]
The ask is to mark `-ODataId` as `required` for commands that call `PUT or POST */$ref`. Non-`$ref` commands will be address by https://github.com/microsoftgraph/msgraph-metadata/issues/278 (API owner dependent).

**Comment by peombwa** [comment]
Fixed by #1967.
